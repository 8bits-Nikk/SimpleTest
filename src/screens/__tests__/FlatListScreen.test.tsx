import {renderWithNavigation} from '../../helper/renderWithNavigation';
import FlatListScreen, {Data} from '../FlatListScreen';
import {fireEvent, waitFor} from '@testing-library/react-native';

describe('FlatScreen Tests', () => {
  it('should render correctly', () => {
    const tree = renderWithNavigation(<FlatListScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have FlatList', () => {
    const tree = renderWithNavigation(<FlatListScreen />);
    const list = tree.getByTestId('FlatList');
    expect(list).not.toBeNull();
  });

  it('should have 30 Items', () => {
    const tree = renderWithNavigation(<FlatListScreen />);
    const list = tree.getByTestId('FlatList');
    expect(list.props.data.length).toBe(30);
  });

  it('should have Item Name', async () => {
    const tree = renderWithNavigation(<FlatListScreen />);
    const list = tree.getByTestId('FlatList');
    const eventData = {
      nativeEvent: {
        contentOffset: {
          y: 500,
        },
        contentSize: {
          // Dimensions of the scrollable content
          height: 500,
          width: 100,
        },
        layoutMeasurement: {
          // Dimensions of the device
          height: 100,
          width: 100,
        },
      },
    };
    await waitFor(() => {
      fireEvent.scroll(list, eventData);
      const items = tree.getAllByTestId('FlatList-ItemName');
      expect(items.length).toBe(30);
      items.forEach((value, index) => {
        expect(value.props.children).toBe(Data[index].name);
      });
    });
  });

  it('should render Items', async () => {
    const tree = renderWithNavigation(<FlatListScreen />);
    const list = tree.getByTestId('FlatList');
    const eventData = {
      nativeEvent: {
        contentOffset: {
          y: 500,
        },
        contentSize: {
          // Dimensions of the scrollable content
          height: 500,
          width: 100,
        },
        layoutMeasurement: {
          // Dimensions of the device
          height: 100,
          width: 100,
        },
      },
    };

    await waitFor(() => {
      fireEvent.scroll(list, eventData);
      const items = tree.getAllByTestId('FlatList-Item');
      console.log(items.length);
      expect(items.length).toBe(30);
      items.forEach(value => {
        expect(value).not.toBeNull();
        fireEvent.press(value);
      });
    });
  });
});
