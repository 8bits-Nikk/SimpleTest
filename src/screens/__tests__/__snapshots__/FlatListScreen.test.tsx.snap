// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlatScreen Tests should render correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ddd",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "id": 1,
            "name": "Ivy NextShares",
          },
          Object {
            "id": 2,
            "name": "Ottawa Bancorp, Inc.",
          },
          Object {
            "id": 3,
            "name": "Alder BioPharmaceuticals, Inc.",
          },
          Object {
            "id": 4,
            "name": "Principal Healthcare Innovators Index ETF",
          },
          Object {
            "id": 5,
            "name": "Canada Goose Holdings Inc.",
          },
          Object {
            "id": 6,
            "name": "Aralez Pharmaceuticals Inc.",
          },
          Object {
            "id": 7,
            "name": "China Mobile (Hong Kong) Ltd.",
          },
          Object {
            "id": 8,
            "name": "J P Morgan Chase & Co",
          },
          Object {
            "id": 9,
            "name": "Matlin & Partners Acquisition Corporation",
          },
          Object {
            "id": 10,
            "name": "Southern California Edison Company",
          },
          Object {
            "id": 11,
            "name": "Live Nation Entertainment, Inc.",
          },
          Object {
            "id": 12,
            "name": "Wells Fargo & Company",
          },
          Object {
            "id": 13,
            "name": "TransAct Technologies Incorporated",
          },
          Object {
            "id": 14,
            "name": "YY Inc.",
          },
          Object {
            "id": 15,
            "name": "Entertainment Gaming Asia Incorporated",
          },
          Object {
            "id": 16,
            "name": "Two River Bancorp",
          },
          Object {
            "id": 17,
            "name": "MEI Pharma, Inc.",
          },
          Object {
            "id": 18,
            "name": "Cerecor Inc.",
          },
          Object {
            "id": 19,
            "name": "Petrobras Argentina S.A.",
          },
          Object {
            "id": 20,
            "name": "PowerShares DWA Momentum Portfolio",
          },
          Object {
            "id": 21,
            "name": "iShares S&P Global Timber & Forestry Index Fund",
          },
          Object {
            "id": 22,
            "name": "Boot Barn Holdings, Inc.",
          },
          Object {
            "id": 23,
            "name": "Natural Health Trends Corp.",
          },
          Object {
            "id": 24,
            "name": "VictoryShares Emerging Market Volatility Wtd ETF",
          },
          Object {
            "id": 25,
            "name": "Engility Holdings, Inc.",
          },
          Object {
            "id": 26,
            "name": "CECO Environmental Corp.",
          },
          Object {
            "id": 27,
            "name": "Micronet Enertec Technologies, Inc.",
          },
          Object {
            "id": 28,
            "name": "Validea Market Legends ETF",
          },
          Object {
            "id": 29,
            "name": "Alphatec Holdings, Inc.",
          },
          Object {
            "id": 30,
            "name": "Capitala Finance Corp.",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      testID="FlatList"
      viewabilityConfigCallbackPairs={Array []}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 12,
                "justifyContent": "center",
                "marginBottom": 16,
                "marginHorizontal": 16,
                "opacity": 1,
                "padding": 8,
                "paddingVertical": 16,
              }
            }
            testID="FlatList-Item"
          >
            <Text
              testID="FlatList-ItemName"
            >
              Ivy NextShares
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 12,
                "justifyContent": "center",
                "marginBottom": 16,
                "marginHorizontal": 16,
                "opacity": 1,
                "padding": 8,
                "paddingVertical": 16,
              }
            }
            testID="FlatList-Item"
          >
            <Text
              testID="FlatList-ItemName"
            >
              Ottawa Bancorp, Inc.
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 12,
                "justifyContent": "center",
                "marginBottom": 16,
                "marginHorizontal": 16,
                "opacity": 1,
                "padding": 8,
                "paddingVertical": 16,
              }
            }
            testID="FlatList-Item"
          >
            <Text
              testID="FlatList-ItemName"
            >
              Alder BioPharmaceuticals, Inc.
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 12,
                "justifyContent": "center",
                "marginBottom": 16,
                "marginHorizontal": 16,
                "opacity": 1,
                "padding": 8,
                "paddingVertical": 16,
              }
            }
            testID="FlatList-Item"
          >
            <Text
              testID="FlatList-ItemName"
            >
              Principal Healthcare Innovators Index ETF
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 12,
                "justifyContent": "center",
                "marginBottom": 16,
                "marginHorizontal": 16,
                "opacity": 1,
                "padding": 8,
                "paddingVertical": 16,
              }
            }
            testID="FlatList-Item"
          >
            <Text
              testID="FlatList-ItemName"
            >
              Canada Goose Holdings Inc.
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 12,
                "justifyContent": "center",
                "marginBottom": 16,
                "marginHorizontal": 16,
                "opacity": 1,
                "padding": 8,
                "paddingVertical": 16,
              }
            }
            testID="FlatList-Item"
          >
            <Text
              testID="FlatList-ItemName"
            >
              Aralez Pharmaceuticals Inc.
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 12,
                "justifyContent": "center",
                "marginBottom": 16,
                "marginHorizontal": 16,
                "opacity": 1,
                "padding": 8,
                "paddingVertical": 16,
              }
            }
            testID="FlatList-Item"
          >
            <Text
              testID="FlatList-ItemName"
            >
              China Mobile (Hong Kong) Ltd.
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 12,
                "justifyContent": "center",
                "marginBottom": 16,
                "marginHorizontal": 16,
                "opacity": 1,
                "padding": 8,
                "paddingVertical": 16,
              }
            }
            testID="FlatList-Item"
          >
            <Text
              testID="FlatList-ItemName"
            >
              J P Morgan Chase & Co
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 12,
                "justifyContent": "center",
                "marginBottom": 16,
                "marginHorizontal": 16,
                "opacity": 1,
                "padding": 8,
                "paddingVertical": 16,
              }
            }
            testID="FlatList-Item"
          >
            <Text
              testID="FlatList-ItemName"
            >
              Matlin & Partners Acquisition Corporation
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 12,
                "justifyContent": "center",
                "marginBottom": 16,
                "marginHorizontal": 16,
                "opacity": 1,
                "padding": 8,
                "paddingVertical": 16,
              }
            }
            testID="FlatList-Item"
          >
            <Text
              testID="FlatList-ItemName"
            >
              Southern California Edison Company
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "height": 0,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#ddd",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="Previous_btn"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Previous
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="Next_btn"
    >
      <View
        style={
          Array [
            Object {},
            Object {},
          ]
        }
      >
        <Text
          disabled={true}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#cdcdcd",
              },
            ]
          }
        >
          Next
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
